### PR TITLE
allow more optional config settings

### PR DESCRIPTION
**Pull request by vetinari**
like the WORKER_PLUGIN_HTTP_OAUTH2_TOKENS. First one was requested, others added because we can :)

Signed-off-by: Hanno Hecker <hanno@zalando.de>

### CODE REVIEW

**Comment by mikkeloscar** [comment]
:+1:

File: cluster/manifests/zmon-worker/deployment.yaml [hunk 1]
```diff
@@ -97,6 +97,37 @@ spec:
                     name: zmon-worker-optional-config
                     key: WORKER_PLUGIN_HTTP_OAUTH2_TOKENS
                     optional: true
+            - name: WORKER_PLUGIN_REDIS_PASSWORD
+              valueFrom:
+                  configMapKeyRef:
```
Comment (aermakov-zalando) [inline cluster/manifests/zmon-worker/deployment.yaml]: This should be a `secretKeyRef`.

File: cluster/manifests/zmon-worker/deployment.yaml [hunk 2]
```diff
@@ -97,6 +97,37 @@ spec:
...
+                    optional: true
+            - name: WORKER_PLUGIN_CASSANDRA_USER
+              valueFrom:
+                  configMapKeyRef:
+                    name: zmon-worker-optional-config
+                    key: WORKER_PLUGIN_CASSANDRA_USER
+                    optional: true
+            - name: WORKER_PLUGIN_CASSANDRA_PASS
+              valueFrom:
+                  configMapKeyRef:
```
Comment (aermakov-zalando) [inline cluster/manifests/zmon-worker/deployment.yaml]: Same, should be a secret.

File: cluster/manifests/zmon-worker/deployment.yaml [hunk 3]
```diff
@@ -97,6 +97,37 @@ spec:
...
+                    name: zmon-worker-optional-config
+                    key: WORKER_PLUGIN_CASSANDRA_PASS
+                    optional: true
+            - name: WORKER_PLUGIN_ELASTICSEARCH_URL
+              valueFrom:
+                  configMapKeyRef:
```
Comment (aermakov-zalando) [inline cluster/manifests/zmon-worker/deployment.yaml:121]: Does this include a password? If yes, it should be in a secret.

File: cluster/manifests/zmon-worker/deployment.yaml [hunk 4]
```diff
@@ -97,6 +97,37 @@ spec:
...
+                    name: zmon-worker-optional-config
+                    key: WORKER_PLUGIN_ELASTICSEARCH_URL
+                    optional: true
+            - name: WORKER_PLUGIN_KAIROSDB_URL
+              valueFrom:
+                  configMapKeyRef:
```
Comment (aermakov-zalando) [inline cluster/manifests/zmon-worker/deployment.yaml:127]: Does this include a password? If yes, it should be in a secret.

**Comment by aermakov-zalando** [comment]
👎

File: cluster/manifests/zmon-worker/deployment.yaml (hunk 4 above)
Comment (vetinari) [reply cluster/manifests/zmon-worker/deployment.yaml:127]: no, not supported

File: cluster/manifests/zmon-worker/deployment.yaml (hunk 3 above)
Comment (vetinari) [reply cluster/manifests/zmon-worker/deployment.yaml:121]: it could potentially, but the default auth method should be with `oauth2=True` as parameter

**Comment by aermakov-zalando** [comment]
👍

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**